modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb

```diff
@@ -135,6 +145,42 @@ def fetch_forms_by_uuid(form_uuid)
       def monitor
         @monitor ||= IvcChampva::Monitor.new
       end
+
+      ##
+      # Tracks metrics for the duration between form submission (DB record creation)
+      # and pega callback confirmation
+      #
+      # @param [String] form_uuid The UUID of the form submission
+      # @param [Array<String>] file_names The file names being processed by pega
+      # @param [ActiveRecord::Relation] forms The already-fetched forms collection
+      #
+      def track_submit_to_callback_duration(_form_uuid, file_names, forms)
+        return unless forms.any?
+
+        # Filter for form files (not supporting documents) that match the provided file_names
+        form_files = forms.select do |form|
+          # Check if this form's file_name is in the provided file_names and is not a supporting document
+          file_names.include?(form.file_name) && form.file_name.exclude?('supporting_doc')
+        end
+
+        return unless form_files.any?
+
+        # Calculate and publish metrics for first matching form file
+        form_files.first do |form|
```

Comment:
Is form_files.first an array? Or should it be if (form = form_files.first). Please add the new method to the corresponding spec file.